flink-table/flink-table-common/src/main/java/org/apache/flink/table/descriptors/DescriptorProperties.java

```diff
@@ -257,6 +267,43 @@ public void putIndexedFixedProperties(String key, List<String> subKeys, List<Lis
 		}
 	}
 
+	/**
+	 * Adds an indexed sequence of properties (with sub-properties) under a common key.
+	 * Different with {@link #putIndexedFixedProperties}, this method supports the properties
+	 * value to be null, which would be ignore. The sub-properties should at least have
+	 * one non-null value.
+	 *
+	 * <p>For example:
+	 *
+	 * <pre>
+	 *     schema.fields.0.type = INT, schema.fields.0.name = test
+	 *     schema.fields.1.type = LONG, schema.fields.1.name = test2
+	 *     schema.fields.2.type = LONG, schema.fields.1.expr = test + 1
+	 * </pre>
+	 *
+	 * <p>The arity of each subKeyValues must match the arity of propertyKeys.
+	 */
+	public void putIndexedOptionalProperties(String key, List<String> subKeys, List<List<String>> subKeyValues) {
```

Comment:
just extend the original `putIndexedFixedProperties` to support null?

Reply:
I prefer not, `putIndexedFixedProperties` has more strict check logic, and it can be used not only for table fields and names, we better keep it i think.